### PR TITLE
[TTNN] Improve SDPA Fusing

### PR DESCRIPTION
**Pull request by jserbedzijaTT**
### Ticket
[Link to Github Issue](https://github.com/tenstorrent/tt-mlir/issues/6335)

### Problem description
The SDPA (Scaled Dot-Product Attention) fusing pattern was not handling newer PyTorch decomposition patterns. Specifically:
- Split-scale pattern: `PyTorch 2.9+` applies` sqrt(1/head_dim)` to both `Q` and `K` separately before the matmul, rather than applying `1/sqrt(head_dim)` to the `QK^T` result.
- NaN handling: Newer torch versions emit a where operation after softmax to handle fully-masked rows (replacing NaN with zeros).

The existing TTIR fusing pattern couldn't match these variations, causing SDPA to remain unfused in models using newer torch versions.

### What's changed

- Moved SDPA fusing from TTIRFusing to TTNNFusing pass
- Added support for split-scale pattern where scale is applied to Q and K separately
- Added support for robust softmax with all-masked row handling (where op pattern)
- Updated tests to cover both single-scale and split-scale SDPA patterns

### Checklist
- [x] New/Existing tests provide coverage for changes

### CODE REVIEW

File: lib/Dialect/TTNN/Transforms/TTNNFusing.cpp [hunk 1]
```diff
@@ -451,6 +968,7 @@ class TTNNFusingPass : public impl::TTNNFusingBase<TTNNFusingPass> {
 #ifdef TTMLIR_ENABLE_OPMODEL
     if (enableOpConstraints) {
       patterns.add<RoPEFusing>(&getContext());
+      patterns.add<SDPAFusing>(&getContext());
```
Comment (vmilosevic) [inline lib/Dialect/TTNN/Transforms/TTNNFusing.cpp:970]: :warning: **clang\-diagnostic\-error** :warning:
unknown type name `` SDPAFusing ``; did you mean `` RoPEFusing ``?
```suggestion
      patterns.add<RoPEFusing>(&getContext());
```
